### PR TITLE
Enable Gossip Scoring

### PR DESCRIPTION
**Pull request by ec2**
**Summary of changes**
Changes introduced in this pull request:
- Reintroduces and fixes GossipSub parameters. 

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Comment by cryptoquick** [comment]
Nice! How would this fail if I were to test it?

**Comment by ec2** [comment]
> Nice! How would this fail if I were to test it?

Blocks come in with delay, DoS, etc...

**Review by creativcoder — APPROVED** [review]
@ec2 where does this improve things?

File: node/forest_libp2p/src/gossip_params.rs [hunk 1]
```diff
@@ -19,19 +19,29 @@ fn build_msg_topic_config() -> TopicScoreParams {
...
         time_in_mesh_cap: 1.0,
 
         // deliveries decay after 10min, cap at 100 tx
-        first_message_deliveries_weight: 5.0,
+        first_message_deliveries_weight: 0.5,
         first_message_deliveries_decay: score_parameter_decay(Duration::from_secs(10 * 60)), // 10mins
         // 100 blocks in an hour
         first_message_deliveries_cap: 100.0,
+
+        // Set to 0 because disabled for FIlecoin
```
Comment (cryptoquick) [inline node/forest_libp2p/src/gossip_params.rs:32]: Misspelled

File: node/forest_libp2p/src/behaviour.rs [hunk 2]
```diff
@@ -421,19 +423,29 @@ impl ForestBehaviour {
...
 
         let gossipsub_config = gs_config_builder.build().unwrap();
-        let gossipsub = Gossipsub::new(
+        let mut gossipsub = Gossipsub::new(
             MessageAuthenticity::Signed(local_key.clone()),
             gossipsub_config,
         )
         .unwrap();
 
         // TODO: Figure out why this delays incoming blocks. See gossip_params.rs for the params settings.
```
Comment (cryptoquick) [inline node/forest_libp2p/src/behaviour.rs:442]: Remove

File: node/forest_libp2p/src/behaviour.rs [hunk 3]
```diff
@@ -421,19 +423,29 @@ impl ForestBehaviour {
...
 
     pub fn new(local_key: &Keypair, config: &Libp2pConfig, network_name: &str) -> Self {
+        let msg_id_fn = |msg: &GossipsubMessage| {
+            let s = blake2b_256(&msg.data);
+            MessageId::from(s)
+        };
         let mut gs_config_builder = GossipsubConfigBuilder::default();
         gs_config_builder.max_transmit_size(1 << 20);
         gs_config_builder.validation_mode(ValidationMode::Strict);
+        gs_config_builder.message_id_fn(msg_id_fn);
```
Comment (cryptoquick) [inline node/forest_libp2p/src/behaviour.rs:433]: Inline closure

File: node/forest_libp2p/src/gossip_params.rs [hunk 4]
```diff
@@ -19,19 +19,29 @@ fn build_msg_topic_config() -> TopicScoreParams {
         topic_weight: 0.1,
 
         // 1 tick per second, maxes at 1 after 1 hour
-        time_in_mesh_weight: 0.00027,
+        time_in_mesh_weight: 0.0002778,
```
Comment (cryptoquick) [inline node/forest_libp2p/src/gossip_params.rs:22]: Leave a comment for `// 1/3600`

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Just some minor changes

**Review by cryptoquick — APPROVED** [review]
Awesome! Hopefully this will reduce those warnings we sometimes get, and just generally make us better at scoring peers with libp2p. :rocket:

**Pull request merged**